superset/tasks/schedules.py

```diff
@@ -538,16 +543,15 @@ def schedule_alert_query(  # pylint: disable=unused-argument
             logger.info("Ignoring deactivated alert")
             return
 
+        if schedule.sql_observers:
+            sql = schedule.sql_observers[0].sql
+
         if report_type == ScheduleType.alert:
             if recipients or slack_channel:
-                deliver_alert(schedule.id, recipients, slack_channel)
+                deliver_alert(schedule.id, sql, recipients, slack_channel)
```

Comment:
how can you deliver alert before checking it?
maybe rename the functions to properly represent what they are doing

it says that both deliver_alert and check_alert notify the customer. slightly confused here.